assets/tests/QuestionSchemaValidationService.js

```diff
@@ -0,0 +1,85 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for validating non-task data put into a question.
+ */
+
+tie.factory('QuestionSchemaValidationService', [
+  'ALL_SUPPORTED_LANGUAGES', 'CLASS_NAME_AUXILIARY_CODE',
+  'CodeCheckerService',
+  function(
+      ALL_SUPPORTED_LANGUAGES, CLASS_NAME_AUXILIARY_CODE,
+      CodeCheckerService) {
+    return {
```

Comment:
Need tests to verify that the BuggyOutputTest messages are unique across all tasks.
Should verify that AuxiliaryCode contains the AuxiliaryCode class definition.
Should probably verify that starter code contains Python function definitions (and we have the regex to validate ... somewhere).
It's probably not a BAD idea to verify that the starter code matches a language pattern for the language specified, but that seems hard so that seems like a "later" thing.

Other than that, this should be fine.